lib/widgets/search_widget.dart

```diff
@@ -0,0 +1,106 @@
+import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:flutter/gestures.dart';
+import 'package:flutter/material.dart';
+import 'package:pso_messanger/configuration/app_text.dart';
+import '../widgets/search_tile_widget.dart';
+import '../configuration/app_path.dart';
+import '../services/database.dart';
+import '../configuration/app_colors.dart';
+import '../widgets/appbar_widget.dart';
+
+class Search extends StatefulWidget {
+  @override
+  _SearchState createState() => _SearchState();
+}
+
+class _SearchState extends State<Search> {
+  DatabaseMethods databaseMethods = DatabaseMethods();
+  TextEditingController searchTextEdittingController = TextEditingController();
+
+  QuerySnapshot searchSnapshot;
+
+  initiateSearch() {
+    databaseMethods
+        .getUserByUserName(searchTextEdittingController.text)
+        .then((value) {
+      setState(() {
+        searchSnapshot = value;
+      });
+    });
+  }
+
+  createChatroomAndStartConversation(String userName) {
+    //List<String> users = [userName, ];
+    //databaseMethods.createChatRoom()
+  }
+
+  Widget _buildSearchList() {
+    return searchSnapshot != null
+        ? ListView.builder(
+            itemCount: searchSnapshot.documents.length,
+            shrinkWrap: true,
+            itemBuilder: (context, index) {
+              return SearchTile(
+                userName: searchSnapshot.documents[index].data["name"],
+                userEmail: searchSnapshot.documents[index].data["email"],
+              );
+            })
+        : Container();
+  }
+
+  @override
+  void initState() {
+    super.initState();
```

Comment:
if you don't initialize anything in initState() you can safely delete this method from this place